packages/server/graphql/private/mutations/updateEmail.ts

```diff
@@ -15,6 +16,39 @@ const updateEmail: MutationResolvers['updateEmail'] = async (_source, {oldEmail,
     throw new Error(`User with ${oldEmail} not found`)
   }
 
+  // Update isEmailVerified to false for the old email
+  // Part 1: Get identities for the user from db
+  const currentIdentitiesQuery = await pg
+    .selectFrom('User')
+    .select('identities')
+    .where('email', '=', oldEmail)
+    .limit(1)
+    .executeTakeFirst()
+
+  if (!currentIdentitiesQuery) {
+    throw new Error(`User with ${oldEmail} not found`)
+  }
+
+  const identitiesArray = currentIdentitiesQuery.identities
+
+  // Part 2: Update the last element of identities array
+  if (identitiesArray && identitiesArray.length > 0) {
+    const lastIndex = identitiesArray.length - 1
+    const lastElement = identitiesArray[lastIndex] as JsonObject;
+    lastElement!.isEmailVerified = false
+  } else {
+    throw new Error('Empty Identities array!')
+  }
+
+  // Part 3: Update the identities in the database
+  await pg
+    .updateTable('User')
+    .set({
+      identities: identitiesArray,
+    })
+    .where('email', '=', oldEmail)
+    .execute()
```

Comment:
-1 instead of having a separate update call, we should update the identities array in the `updateTable('User')` call further below. We should also find the identity matchin `AuthIdentityLocal` and update that one explicitly.

Reply:
Sure, updated the PR. Please have a second look.